tasks.py

```diff
@@ -146,3 +119,86 @@ def parse_spec_for_filters(_):
     with open(optional_filter_path, "w") as f:
         for _filter in optional_filters:
             f.write(_filter + "\n")
+
+
+@task
+def generate_openapi(_):
+    """Update OpenAPI schema in file 'local_openapi.json'"""
+    from optimade.server.main import app
+
+    if not TOP_DIR.joinpath("openapi").exists():
+        os.mkdir(TOP_DIR.joinpath("openapi"))
+    with open(TOP_DIR.joinpath("openapi/local_openapi.json"), "w") as f:
+        json.dump(app.openapi(), f, indent=2)
+
+    """Update OpenAPI schema in file 'local_index_openapi.json'"""
+    from optimade.server.main_index import app as app_index
+
+    if not TOP_DIR.joinpath("openapi").exists():
+        os.mkdir(TOP_DIR.joinpath("openapi"))
+    with open(TOP_DIR.joinpath("openapi/local_index_openapi.json"), "w") as f:
+        json.dump(app_index.openapi(), f, indent=2)
+
+
+@task(generate_openapi)
+def check_openapi_diff(_):
+    """Checks the Generated OpenAPI spec against what is stored in the repo"""
+    with open(TOP_DIR.joinpath("openapi/openapi.json")) as f:
+        openapi = json.load(f)
+
+    with open(TOP_DIR.joinpath("openapi/local_openapi.json")) as f:
+        local_openapi = json.load(f)
+
+    with open(TOP_DIR.joinpath("openapi/index_openapi.json")) as f:
+        index_openapi = json.load(f)
+
+    with open(TOP_DIR.joinpath("openapi/local_index_openapi.json")) as f:
+        local_index_openapi = json.load(f)
+
+    openapi_diff = diff(openapi, local_openapi)
+    if openapi_diff != {}:
+        print(
+            "Generated OpenAPI spec for test server did not match committed version.\n"
+            "Run 'invoke update-openapijson' and re-commit.\n"
+            f"Diff:\n{openapi_diff}"
+        )
+        sys.exit(1)
+
+    openapi_index_diff = diff(index_openapi, local_index_openapi)
+    if openapi_index_diff != {}:
+        print(
+            "Generated OpenAPI spec for Index meta-database did not match committed version.\n"
+            "Run 'invoke update-openapijson' and re-commit.\n"
+            f"Diff:\n{openapi_index_diff}"
+        )
+        sys.exit(1)
+
+
+@task(generate_openapi)
+def update_openapijson(c):
+    """Updates the stored OpenAPI spec to what the server returns"""
+    # pylint: disable=import-outside-toplevel
+    c.run(
+        f"cp {TOP_DIR.joinpath('openapi/local_openapi.json')} {TOP_DIR.joinpath('openapi/openapi.json')}"
+    )
+    c.run(
+        f"cp {TOP_DIR.joinpath('openapi/local_index_openapi.json')} {TOP_DIR.joinpath('openapi/index_openapi.json')}"
+    )
+
+
+@task
+def update_api_shield(_):
+    """Updates the Github OPTIMADE API Shield"""
+    shield = {
+        "schemaVersion": 1,
+        "label": "OPTIMADE",
+        "message": __api_version__,
```

Comment:
Do we want this? See same question above.
```suggestion
        "message": f"v{__api_version__}",
```

Reply:
As in, do we want to remove generating this shield? I'm completely agnostic to the logo. Don't think it adds anything, but it also doesn't hurt.

Reply:
nooo, the shield is jam-tastic! It was more the thing of this PR changing the version label from `v0.10.1` -> `0.10.1`, i.e., losing the `v`.
Personally, I would prefer to keep the `v`, to be similar to the PyPI badge's label.